### PR TITLE
Admin permission detection of Role classe name

### PR DESCRIPTION
**Pull request by 123monsite-regis**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | When a back office tab is created, Roles and permissions depends of php class name of the tab, as module can  create them also, the check should accept them
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | see bellow please

-----------
How to test
Create a module Who register a BO tab, ( an example can be found on  ps_linklist : https://github.com/PrestaShop/ps_linklist/blob/master/ps_linklist.php#L87 )
name the tab class for instances AdminLink2_Widget , 

then look at your bo menu, , before this pach, permissions could not be checked, so only super profile had access to the tab.

----------
Where i find this rexep ?
this regexp came from Php docs, and seem to be the recommanded rexexp to check a Class name
http://php.net/manual/en/language.oop5.basic.php

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8700)

### CODE REVIEW

**Comment by LittleBigDev** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8700#-:-L418VA0pLlvyBZgX4kx:bnfp4nl)*

**Review by LittleBigDev — DISMISSED** [review]
Thank you @123monsite-regis 
Ok for me.

File: classes/Profile.php [hunk 1]
```diff
@@ -214,7 +214,7 @@ private static function fillCacheAccesses($idProfile, $type, $cacheData = [])
     private static function findIdTabByAuthSlug($authSlug)
     {
         preg_match(
-            '/ROLE_MOD_[A-Z]+_(?P<classname>[A-Z]+)_(?P<auth>[A-Z]+)/',
+            '/ROLE_MOD_[A-Z]+_(?P<classname>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)_(?P<auth>[A-Z]+)/',
```
Comment (eternoendless) [inline classes/Profile.php]: I'm not sure about the `\x7f-\xff` part.

While theoretically speaking "¼¸¬" would be a valid className, I'm not sure if it's a good idea to allow those characters there.

In addition, why did you change the `+` into `*`? That allows `ROLE_MOD_FOO__BAR`.

Comment (123monsite-regis) [reply classes/Profile.php]: I'm not sure about the \x7f-\xff part, mmm php accept it,  http://php.net/manual/en/language.oop5.basic.php , Œ is part of this range 

about + into * ? , it is not really change the regular expression has now tow parts, first car = [a-zA-Z_\x7f-\xff] (a valid class name can't start whit a number) and, rest of classname = [a-zA-Z0-9_\x7f-\xff]* , so at minimum one car is required

after reading again the expression, i constat that i forgot that a role must be Uppercase,
`/ROLE_MOD_[A-Z]+_(?P<classname>[A-Z][A-Z0-9]*)_(?P<auth>[A-Z]+)/`
may be the best choice here and cover all the possibilities exept  \x7f-\xff and underscores inside class names

Comment (eternoendless) [reply classes/Profile.php]: > I'm not sure about the \x7f-\xff part, mmm php accept it, http://php.net/manual/en/language.oop5.basic.php , Œ is part of this range

Yeah, that's why I spoke about "theoretically valid class names". What was actually troubling me was the possible unforeseen side effects of allowing non-word characters in the roles.

You're right about the `*` though, I got mixed up when reading the regex 👍

**Review by eternoendless — APPROVED** [review]
Looks good to me!

**Comment by LittleBigDev** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8700)*

**Review by LittleBigDev — APPROVED** [review]

**Comment by marionf** [comment]
Hello @123monsite-regis 
Does it also solve this issue: http://forge.prestashop.com/browse/BOOM-3669 ?

**Comment by 123monsite-regis** [comment]
hello, 
sorry not,
the issue is not similar.
On the forge ticket, it seem to appear on an uprgaded shop; it does not occur on a fresh shop.

**Comment by eternoendless** [comment]
Thank you @123monsite-regis
